### PR TITLE
feat: put enums on their own page in HTML documentation

### PR DESCRIPTION
**Pull request by sockmaster27**
Related to #6327

### CODE REVIEW

**Comment by sockmaster27** [comment]
I'd like to refactor this part so it's more consistent:
```
  /**
    * A represention of an enum that's easier to work with while generating documention.
    */
  private case class Enum(enm: TypedAst.Enum,
                          parent: Symbol.ModuleSym,
                          companionMod: Option[Module])

  /**
    * A represention of a class that's easier to work with while generating documention.
    */
  private case class Class(doc: Ast.Doc,
                           ann: Ast.Annotations,
                           modifiers: Ast.Modifiers,
                           sym: Symbol.ClassSym,
                           parent: Symbol.ModuleSym,
                           tparam: TypedAst.TypeParam,
                           superClasses: List[Ast.TypeConstraint],
                           assocs: List[TypedAst.AssocTypeSig],
                           signatures: List[TypedAst.Sig],
                           defs: List[TypedAst.Sig],
                           laws: List[TypedAst.Def],
                           instances: List[TypedAst.Instance],
                           companionMod: Option[Module],
                           loc: SourceLocation)
```

Do you have a preference? As I see it, wrapping the `TypedAst` node makes it less coupled and reduces the amount of code in this file, but is a bit more awkward to work with (a lot of `enm.enm.sym`)

**Comment by magnus-madsen** [comment]
I would prefer to reuse TypedAst as much as possible.

File: main/test/flix/Test.Exp.CheckedTypeCast.flix
Comment (sockmaster27) [inline main/test/flix/Test.Exp.CheckedTypeCast.flix:1]: Something weird is going on here

**Comment by magnus-madsen** [comment]
!!!jar

File: main/src/ca/uwaterloo/flix/language/phase/HtmlDocumentor.scala [hunk 1]
```diff
@@ -1045,33 +1133,25 @@ object HtmlDocumentor {
...
                            instances: List[TypedAst.Instance],
-                           companionMod: Option[Module],
-                           loc: SourceLocation) extends MainItem
+                           parent: Symbol.ModuleSym,
+                           companionMod: Option[Module])
+
+  /**
+    * A represention of an enum that's easier to work with while generating documention.
+    */
+  private case class Enum(inner: TypedAst.Enum,
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/HtmlDocumentor.scala:1154]: Rename `inner` to `decl` (to be consistent with the rest of the compiler).

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Lets get types and effects done and then we can iterate. Ill merge this in once tests complete.

**Pull request merged**